substrate/client/authority-discovery/src/worker.rs

```diff
@@ -308,20 +339,42 @@ where
 			warn_public_addresses: false,
 			phantom: PhantomData,
 			last_known_records: HashMap::new(),
+			spawner,
+			persisted_cache_file_path: maybe_persisted_cache_file_path,
 		}
 	}
 
+	/// Persists `AddrCache` to disk if the `persisted_cache_file_path` is set.
+	pub fn persist_addr_cache_if_supported(&self) {
+		let Some(path) = self.persisted_cache_file_path.as_ref().cloned() else {
+			return;
+		};
+		let Some(serialized_cache) = self.addr_cache.serialize() else { return };
+		self.spawner.spawn(
+			"PersistAddrCache",
+			Some("AuthorityDiscoveryWorker"),
+			Box::pin(async move { AddrCache::persist(path, serialized_cache) }),
+		)
+	}
+
 	/// Start the worker
 	pub async fn run(mut self) {
+		let mut persist_interval =
+			tokio::time::interval(Duration::from_secs(60 * 10 /* 10 minutes */));
```

Comment:
Would be good to place this in a constant at the beginning of the file so we can adjust it easily and see which constants we take into account in this module (maybe we'll need to modify it later)